src/veryfi/Client.php

```diff
@@ -550,4 +551,297 @@ public function replace_tags(int $document_id,
+     * @param boolean $confidence_details Return the score and OCR score fields in the document response
+     * @return string Object of previously processed bank statements
+     */
+    public function get_bank_statements(int $page = 1, int $page_size = 50, bool $bounding_boxes = false, bool $confidence_details = false): string
+    {
+        $endpoint_name = "/bank-statements/";
+        $request_arguments = [
+            'page' => $page,
+            'page_size' => $page_size,
+            'bounding_boxes' => $bounding_boxes,
+            'confidence_details' => $confidence_details
+        ];
+        return $this->request("GET", $endpoint_name, $request_arguments);
+    }
+
+    /**
+     * Process a bank statement from a file path and extract all fields from it.
+     *
+     * @param string $file_path Path on disk to a file to submit for data extraction.
+     * @param bool $bounding_boxes Return bounding box and bounding region for extracted fields.
+     * @param bool $confidence_details Return the score and OCR score fields in the document response.
+     * @param array $additional_request_parameters Additional request parameters.
+     * @return string Data extracted from the document.
+     */
+    public function process_bank_statement(string $file_path,
+                                           bool $bounding_boxes = false,
+                                           bool $confidence_details = false,
+                                           array $additional_request_parameters = array()): string
+    {
+        $endpoint_name = '/bank-statements/';
+        $file_name = pathinfo($file_path, PATHINFO_BASENAME);
+        $file = fopen($file_path, 'rb');
+        $file_data = base64_encode(fread($file, filesize($file_path)));
+        $request_arguments = array(
+            'file_name' => $file_name,
+            'file_data' => $file_data,
+            'bounding_boxes' => $bounding_boxes,
+            'confidence_details' => $confidence_details
+        );
+        $request_arguments = array_replace($request_arguments, $additional_request_parameters);
+        return $this->request('POST', $endpoint_name, $request_arguments);
+    }
+
+    /**
+     * Process a bank statement and extract all fields from it.
+     *
+     * @param string $file_path Path on disk to a file to submit for data extraction
+     * @param boolean $bounding_boxes Return bounding box and bounding region for extracted fields
+     * @param boolean $confidence_details Return the score and OCR score fields in the document response
+     * @param array $additional_request_parameters Additional request parameters
+     * @return string Data extracted from the document
+     */
+    public function process_bank_statement_from_file(string $file_path, bool $bounding_boxes = false, bool $confidence_details = false, array $additional_request_parameters = []): string
```

Comment:
same as process_bank_statement?

Reply:
Done I removed the function who uses file_get_contents() because it can consume more memory for very large files since it loads everything into memory at once